code/game/objects/objs.dm

```diff
@@ -48,6 +48,11 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 	if(src in processing_objects)
 		processing_objects -= src
 
+	if(integratedpai)
+		integratedpai.forceMove(get_turf(src))
```

Comment:
I guess it'll just be left up to the individual coder, when they make an object integrative, to add code to auto-eject the pAI where necessary.